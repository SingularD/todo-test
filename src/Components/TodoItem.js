import React from 'react'

class TodoItem extends React.Component{
  constructor(props) {
    super(props)
  }
  addDoneItem(index) {
    this.props.addDoneItem(index)
  }
  removeDoneItem(index) {
    this.props.removeDoneItem(index)
  }
  render() {
    const { type, content, index } = this.props
    return (
      <li className="list-group-item">
        <div className="row justify-content-between">
          <div className="col-6 font-weight-bold content">{content}</div>
          <div className="col-4 row justify-content-center">
            {
              type === 'todo' ?
                <button
                  className="btn btn-primary col"
                  onClick={this.addDoneItem.bind(this, index)}
                >
                  Done
                </button> :
                <button
                  className="btn btn-danger col"
                  onClick={this.removeDoneItem.bind(this, index)}
                >
                  Delete
                </button>
            }
          </div>
        </div>
      </li>
    )
  }
}
export default TodoItem
