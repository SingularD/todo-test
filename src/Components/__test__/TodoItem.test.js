import React from 'react'
import { shallow } from 'enzyme'
import TodoItem from '../TodoItem'

const props = {
  type:"todo",
  content: "this is a todoItem",
  index: 1,
  addDoneItem: jest.fn()
}

describe('测试todoItem和doneItem', () => {
  let wrapper = shallow(<TodoItem {...props}/>)
  it('匹配snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it("正确地展示内容", () => {
    expect(wrapper.find('.content').text()).toEqual('this is a todoItem')
    expect(wrapper.instance().props.index).toEqual(1)
  })
  it('点击按钮触发函数', () => {
    wrapper.find('button').simulate('click')
    expect(props.addDoneItem).toHaveBeenCalledWith(1)
  })
})
