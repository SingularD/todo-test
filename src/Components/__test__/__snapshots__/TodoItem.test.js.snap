// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试todoItem和doneItem 匹配snapshot 1`] = `ShallowWrapper {}`;
